cmd/esc/cli/login.go

```diff
@@ -157,7 +154,33 @@ func (esc *escCommand) getCachedClient(ctx context.Context) error {
 		return fmt.Errorf("could not determine current cloud: %w", err)
 	}
 	if account == nil {
-		return fmt.Errorf("no credentials. Please run `%v login` to log in.", esc.command)
+		backendURL := esc.workspace.GetCurrentCloudURL(nil)
+
+		nAccount, err := esc.login.Login(
+			ctx,
+			backendURL,
+			false,
+			"esc",
+			"Pulumi ESC environments",
+			nil,
+			false,
+			display.Options{Color: esc.colors},
+		)
```

Comment:
I think the only remaining question I have is what the experience is like if no access token is specified via `PULUMI_ACCESS_TOKEN`? If we fall through to the `err != nil` case below I think that we'll be regressing the UX, as the user will see `could not determine current cloud` instead of `no credentials. Please run esc login to log in.`

Reply:
It is the same experience you have with the pulumi cli where it asks you to authenticate

<img width="506" alt="image" src="https://github.com/pulumi/esc/assets/5647310/4903521b-1a0a-4b08-9f5d-d32cda28c844">

<img width="493" alt="image" src="https://github.com/pulumi/esc/assets/5647310/72ffb3b6-3e38-442e-9b6b-51636dfe031d">

Reply:
excellent; thanks!

Reply:
what about in non-interactive scenarios?

Reply:
(try running with the output redirected, for example)

Reply:
This is the output
<img width="843" alt="image" src="https://github.com/pulumi/esc/assets/5647310/bf1ab061-7b01-4e1b-995f-7d61204742f6">